### PR TITLE
Properly set worker_query and use instead of application name on `citus_[dist/worker]_stat_activity`

### PR DESCRIPTION
**Pull request by onderkalaci**
Relying on `pg_stat_activity.application_name` is not useful because if we don't have enough privileges, `pg_stat_activity.application_name == NULL` shown as NULL.

That's why, we properly use `worker_query` field and rely on that again.

This is a step for proper `citus_[extended]_stat_activity` implementation.

### CODE REVIEW

File: src/backend/distributed/test/run_from_same_connection.c
Comment (onderkalaci) [inline src/backend/distributed/test/run_from_same_connection.c:192]: I'll squash the last 4 commits into one

File: src/backend/distributed/test/run_from_same_connection.c
Comment (halilozanakgul) [inline src/backend/distributed/test/run_from_same_connection.c:189]: The function name is incorrect

File: src/backend/distributed/test/run_from_same_connection.c
Comment (halilozanakgul) [inline src/backend/distributed/test/run_from_same_connection.c:127]: Why is this called `assignGPID`?

File: src/backend/distributed/transaction/backend_data.c
Comment (halilozanakgul) [inline src/backend/distributed/transaction/backend_data.c:831]: Can you add a test to set an example?

File: src/test/regress/spec/isolation_mx_common.include.spec
Comment (halilozanakgul) [inline src/test/regress/spec/isolation_mx_common.include.spec:16]: Does this not need the `$$override_backend_data_command_originator$$` part?

File: src/backend/distributed/transaction/backend_data.c
Comment (halilozanakgul) [inline src/backend/distributed/transaction/backend_data.c:835]: This is for overriding the `distributedCommandOriginator` which is the opposite of `WorkerQuery`.
Can you rename to `OverrideBackendDataDistributedCommandOriginator`?

File: src/backend/distributed/transaction/backend_data.c
Comment (halilozanakgul) [inline src/backend/distributed/transaction/backend_data.c:416]: Does this name still work?

File: src/backend/distributed/transaction/backend_data.c
Comment (halilozanakgul) [inline src/backend/distributed/transaction/backend_data.c:680]: I'm a bit concerned with this unsetting. We make it false, which means this is a worker query.
Looks like we used to do this way too, but maybe we could convert to a bool pointer for true, false and NULL, or maybe a int with enums `ORIGINATOR_QUERY`, `WORKER_QUERY` and `UNASSIGNED`.

I'm not sure if this is an actual problem though 😄

**Review by halilozanakgul — CHANGES_REQUESTED** [review]

File: src/backend/distributed/transaction/backend_data.c
Comment (onderkalaci) [reply src/backend/distributed/transaction/backend_data.c:835]: Ops, there is a confusion between WorkerQuery and distributedCommandOriginator. I'll make all distributedCommandOriginator.

File: src/backend/distributed/transaction/backend_data.c
Comment (onderkalaci) [reply src/backend/distributed/transaction/backend_data.c:831]: (It might be that isolation tests do not show up in coverage reports, I vaguely remember this)

File: src/test/regress/spec/isolation_mx_common.include.spec
Comment (onderkalaci) [reply src/test/regress/spec/isolation_mx_common.include.spec:16]: not necessarily, by default PG searches for the same name already.

Having `$$` inside `run_command_on_workers` giving weird syntax errors, hence skipped that (also didn't seem important as this is test-only function)

File: src/backend/distributed/transaction/backend_data.c
Comment (onderkalaci) [reply src/backend/distributed/transaction/backend_data.c:680]: well, we don't even need to set this. I'd rather remove :D

**Review by halilozanakgul — APPROVED** [review]

**Comment by onderkalaci** [comment]
this is on top of #5702, so I'll merge once we merge that

**Pull request merged**
